rtos/rtx/TARGET_CORTEX_M/rt_CMSIS.c

```diff
@@ -302,7 +302,7 @@ static inline  t __##f (t1 a1, t2 a2, t3 a3, t4 a4) {                          \
 #define SVC_Setup(f)                                                           \
   __asm(                                                                       \
     "mov r12,%0\n"                                                             \
-    :: "r"(&f): "r12"                                                          \
+    :: "r"(&f): "r0", "r1", "r2", "r3", "r12"                                  \
```

Comment:
It is emitting an IAR specific __swi function declaration and call for each svc, so the compiler should be able to handle the memory ordering appropriately. The issue is that the functions are emitted with fewer arguments than are clobbered in the SVC handler.
``` c
_Pragma("swi_number=0") __swi t _##f (t1 a1);
```